### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Mu7ammadAbed**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
+import { BrowserRouter as Router } from 'react-router-dom';
+import NavBar from './components/Navbar';
 import { Footer } from './components';
 
 function App() {
   return (
     <div className="App">
-      <Footer />
+      <Router>
```
Comment (NoureldeanSaed) [inline client/src/App.js:8]: This isn't how `Router` is used. Check the documentation.

File: client/src/components/Navbar/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        </Link>
+        {' '}
+        <List
+          className="list"
+          sx={{ justifyContent: 'space-between' }}
+          style={{
+            marginRight: '150px', alignSelf: 'center', color: '#797B7D',
+          }}
+        >
+          <Link to="/" style={{ color: '#797B7D', textDecoration: 'none' }}>
```
Comment (NoureldeanSaed) [inline client/src/components/Navbar/index.js:28]: Try and make all of this
```jsx
<Link to="/signup" style={{ color: '#797B7D', textDecoration: 'none' }}>
  <ListItem button>
    <ListItemText primary="About" />
  </ListItem>
</Link>
```
Into one component, and reuse it here as much as you'd like.

File: client/src/components/Navbar/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import {
+  ListItem, ListItemText, Button, Avatar, Container, List,
+} from '@mui/material';
+import './style.css';
+
+import Logo from '../../asstes/logo.png';
+import PresonImg from '../../asstes/avatar.png';
+
+function NavBar() {
+  const [logged, setLogged] = useState(true);
```
Comment (NoureldeanSaed) [inline client/src/components/Navbar/index.js:13]: Why do you have this state?
Answer:
> We are going to implement the login behavior for later.
>> Reply: You shouldn't, all user data/authentication/authorization/permissions should be stored in a singular store/context and fetched only from there.

File: server/database/quieres/filterEstates.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const connection = require('../config/connection');
+
+module.exports = ({
+  type, category, location, price, roomNumbers, bathRooms, space,
+}) => {
+  let text = 'SELECT * FROM estates WHERE ';
+  const values = [];
+
+  if (type) {
+    text += `type = $${values.length + 1} AND `;
```
Comment (NoureldeanSaed) [inline server/database/quieres/filterEstates.js:10]: Never in your life use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for SQL queries.

Comment (NoureldeanSaed) [reply server/database/quieres/filterEstates.js:10]: Also, one piece of advice for everyone @Mu7ammadAbed @muhammadharoun @ElhamFadel @mohammedsalah7, especially those who intend to specialize in Backend, you *MUST* at least once check a security course.

**Review by NoureldeanSaed — CHANGES_REQUESTED** [review]

File: server/database/quieres/filterEstates.js (hunk 4 above)
Comment (ElhamFadel) [reply server/database/quieres/filterEstates.js:10]: Can you suggest us one? @NoureldeanSaed

**Pull request merged**